### PR TITLE
feat: new hook `useMeasure`

### PR DESCRIPTION
**Pull request by xobotyi**
## What new hook does?

 Uses ResizeObserver to track element dimensions and re-render component when they change.

## Checklist

- [x] Have you read [contribution guideline](../../CONTRIBUTING.md)?
- [x] Does the code have comments in hard-to-understand areas?
- [x] Is there an existing issue for this PR?
  - #33 
- [x] Have the files been linted and formatted?
- [x] Have the docs been updated?
- [x] Have the tests been added to cover new hook?
- [x] Have you run the tests locally to confirm they pass?

### CODE REVIEW

**Comment by JoeDuncko** [comment]
Why not just call this "useResize"?

File: stories/Sensor/useMeasure.story.mdx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+import { Example } from './useMeasure.stories';
+
+<Meta title="Sensor/useMeasure" />
+
+# useMeasure
+
+Uses ResizeObserver to track element dimensions and re-render component when they change.
+
+- It's ResizeObserver callback uses RAF debouncing, therefore it is pretty performant.
```
Comment (JoeDuncko) [inline stories/Sensor/useMeasure.story.mdx:10]: its*

File: stories/Sensor/useMeasure.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { useMeasure } from '../../src/useMeasure';
+
+export const Example: React.FC = () => {
+  const [measure, ref] = useMeasure<HTMLDivElement>();
```
Comment (JoeDuncko) [inline stories/Sensor/useMeasure.stories.tsx:5]: I think `measurements` might be more descriptive than `measure`.

**Review by JoeDuncko — APPROVED** [review]

**Comment by xobotyi** [comment]
> Why not just call this "useResize"?

It does not resize anything? 🤔
I also dont really like `useMeasure`, but this name is quite widespreaded and looks like most suitable for what it does.

**Pull request merged**